src/gui/editors/SongEditor.cpp

```diff
@@ -837,17 +832,49 @@ void SongEditor::updatePositionLine()
 
 
 
+//! Convert zoom slider's value to bar width in pixels
+int SongEditor::calculatePixelsPerBar() const
+{
+	// What we need to raise 2 by to get MIN_PIXELS_PER_BAR and MAX_PIXELS_PER_BAR
+	double minExp = log2(MIN_PIXELS_PER_BAR);
+	double maxExp = log2(MAX_PIXELS_PER_BAR);
+	double exponent = m_zoomingModel->value() / double(ZOOM_STEPS) * (maxExp - minExp) + minExp;
+
+	double ppb = pow(2, exponent);
+
+	return static_cast<int>(std::round(ppb));
+}
+
+
+
+
+//! Convert bar width in pixels to zoom slider value
+int SongEditor::calculateZoomSliderValue(int pixelsPerBar) const
+{
+	// What we need to raise 2 by to get MIN_PIXELS_PER_BAR and MAX_PIXELS_PER_BAR
+	double minExp = log2(MIN_PIXELS_PER_BAR);
+	double maxExp = log2(MAX_PIXELS_PER_BAR);
+	double exponent = log2(pixelsPerBar);
```

Comment:
```suggestion
	static const double minExp = std::log2(MIN_PIXELS_PER_BAR);
	static const double maxExp = std::log2(MAX_PIXELS_PER_BAR);
	double exponent = std::log2(pixelsPerBar);
```
Can use `static const` local variables because they only need to be calculated once.
Added `std::` prefix to math functions.